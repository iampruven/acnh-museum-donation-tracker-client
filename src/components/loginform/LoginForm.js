import React from "react";
import './loginForm.css'
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service'
export default class LoginForm extends React.Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = {
    username: { value: "" },
    password: { value: "" },
    error: null
  };

  handleSubmitJwtAuth = (ev)=>{
    ev.preventDefault();
    this.setState({error:null})
    const {username,password} =ev.target
    AuthApiService.postLogin({
      username:username.value,
      password:password.value
    })
    .then(res=>{
      username.value=""
      password.value=""
      TokenService.saveAuthToken(res.authToken)
      this.props.onLoginSuccess()
      
    })
    .catch(res=>{
      this.setState({error:res.error})
    })
  }
  

  updateUsername(username) {
    this.setState({ username: { value: username } });
  }
  updatePassword(password) {
    this.setState({ password: { value: password } });
  }

  render() {
    const { error } = this.state
    return (
      <div className="backdrop">
        <h1 className="centered">Login</h1>
        <form className="centered column" onSubmit={this.handleSubmitJwtAuth}>
          <label htmlFor="username">Username</label>
          <input
            id="username"
            type="text"
            name="username"
            placeholder="username"
            value={this.state.username.value}
            onChange={(e) => this.updateUsername(e.target.value)}
            required
            autoComplete="off"
            aria-required="true"
            aria-label="Login to the website with your username."
          />
          <br />
          <label htmlFor="password">Password</label>
          <input
            id="password"
            type="password"
            name="password"
            value={this.state.password.value}
            onChange={(e) => this.updatePassword(e.target.value)}
            required
            autoComplete="off"
            aria-required="true"
            aria-label="Login to the website with your password."
          />
          <br />
          <button className="btn" type="submit">Submit</button>
        </form>
        <div className="alert">{error}</div>
      </div>
    );
  }
}
