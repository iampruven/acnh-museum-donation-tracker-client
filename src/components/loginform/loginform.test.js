import React from 'react';
import ReactDOM from 'react-dom';
import LoginForm from './LoginForm';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
//smoke test
describe('renders without crashing',()=>{
    it('it renders without crashing',()=>{
        const div = document.createElement('div');

        ReactDOM.render(<MemoryRouter><LoginForm /></MemoryRouter>,div);
        ReactDOM.unmountComponentAtNode(div);
    })


//snapshot test
    it('renders the UI as expected', ()=>{
        const tree = renderer
            .create(<MemoryRouter><LoginForm /></MemoryRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })



})