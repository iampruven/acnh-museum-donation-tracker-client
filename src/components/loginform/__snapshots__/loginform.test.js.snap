// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders the UI as expected 1`] = `
<div
  className="backdrop"
>
  <h1
    className="centered"
  >
    Login
  </h1>
  <form
    className="centered column"
    onSubmit={[Function]}
  >
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      aria-label="Login to the website with your username."
      aria-required="true"
      id="username"
      name="username"
      onChange={[Function]}
      placeholder="username"
      required={true}
      type="text"
      value=""
    />
    <br />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      aria-label="Login to the website with your password."
      aria-required="true"
      id="password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <br />
    <button
      className="btn"
      type="submit"
    >
      Submit
    </button>
  </form>
  <div
    className="alert"
  />
</div>
`;
